File: 0055-jump-game/0055-jump-game.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:  
       maxReach=0
       i=0
       while i <= maxReach:
          maxReach=max(maxReach, i+nums[i])
          i+=1
          if maxReach >= len(nums)-1:
             return True

File: 0055-jump-game/0055-jump-game.java
class Solution {
    public boolean canJump(int[] nums) {
        int maxReach = 0;
        int i = 0;
        while (i <= maxReach) {
            maxReach = Math.max(maxReach, i + nums[i]);
            i++;
            if (maxReach >= nums.length - 1) {
                return true;
            }
        }
        return false;
    }
}